lib/intern.dl

```diff
@@ -14,6 +14,7 @@
  * There is simply no way to obtain an interned object of a different type.
  */
 #[size=4]
+#[shared_ref]
```

Comment:
I don't really grasp what the user should know about this annotation.
I don't really understand what it does; I suspect it's really tied to the Rust implementation.

Reply:
It is. I don't expect DDlog users to use this annotation, but it allows us to introduce new smart pointers in the future without changing the compiler.